core/listener.go

```diff
@@ -160,6 +160,10 @@ func (cl *Listener) handleNewSignedBlock(ctx context.Context, b types.EventDataS
 	// generate extended header
 	eh, err := cl.construct(ctx, &b.Header, &b.Commit, &b.ValidatorSet, eds)
 	if err != nil {
+		var errMismatch *header.ErrDataRootMismatch
+		if errors.As(err, &errMismatch) {
+			panic(err.Error())
+		}
```

Comment:
Why only panic on mismatch? If Bridge gives us invalid header for any other reason panicking makes sense. This would aloes simplify code a bit by removing special node type.